cc/core/Vec.cc

```diff
@@ -39,42 +40,74 @@ NAN_MODULE_INIT(Vec::Init) {
-		return tryCatch.throwError("Vec::New - expected arguments (w), x, y, (z)");
+	if (info.Length() < 2 || info.Length() > 6 || info.Length() == 5) {
+		return tryCatch.throwError("Vec::New - expected arguments (u, v), (w), x, y, (z)");
 	}
 	v8::Local<v8::Object> jsVec;
-	if (info.Length() == 4) {
-		jsVec = FF::newInstance(Nan::New(Vec4::constructor));
-		Nan::ObjectWrap::Unwrap<Vec4>(jsVec)->self = cv::Vec4d(
-			info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
-			info[1]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
-			info[2]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
-			info[3]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value()
-		);
-	} else {
-		double x = info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value();
-		double y = info[1]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value();
-		if (info.Length() == 3) {
-			double z = info[2]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value();
-			jsVec = FF::newInstance(Nan::New(Vec3::constructor));
-			Nan::ObjectWrap::Unwrap<Vec3>(jsVec)->self = cv::Vec3d(x, y, z);
-		}
-		else {
+
+	switch(info.Length()) {
+		case 2:
 			jsVec = FF::newInstance(Nan::New(Vec2::constructor));
-			Nan::ObjectWrap::Unwrap<Vec2>(jsVec)->self = cv::Vec2d(x, y);
-		}
+		  Nan::ObjectWrap::Unwrap<Vec2>(jsVec)->self = cv::Vec2d(
+		  	info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[1]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value()
+		  );
+			break;
+		case 3:
+		  jsVec = FF::newInstance(Nan::New(Vec3::constructor));
+		  Nan::ObjectWrap::Unwrap<Vec3>(jsVec)->self = cv::Vec3d(
+		  	info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[1]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[2]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value()
+		  );
+			break;
+		case 4:
+		  jsVec = FF::newInstance(Nan::New(Vec4::constructor));
+		  Nan::ObjectWrap::Unwrap<Vec4>(jsVec)->self = cv::Vec4d(
+		  	info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[1]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[2]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[3]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value()
+		  );
+			break;
+		case 6:
+		  jsVec = FF::newInstance(Nan::New(Vec6::constructor));
+		  Nan::ObjectWrap::Unwrap<Vec6>(jsVec)->self = cv::Vec6d(
+		  	info[0]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[1]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[2]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[3]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[4]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value(),
+		  	info[5]->ToNumber(Nan::GetCurrentContext()).ToLocalChecked()->Value()
+		  );
```

Comment:
I wanted to rewrite this using macros to minimise the duplication but I'm too unfamiliar in cpp land to do it correctly, so this is the alternative I came up with.

Reply:
No worries, no need to introduce macros here :)